File: Data Structures & Algorithms/remove-element/submission-0.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int k = 0; // Pointer for the next valid element's position
        
        for (int i = 0; i < nums.size(); i++) {
            // If the current element is NOT the one we want to remove
            if (nums[i] != val) {
                nums[k] = nums[i]; // Move it to the front
                k++;               // Move the k pointer forward
            }
        }
        
        // k now represents the total number of elements not equal to val
        return k;
    }
};

File: Data Structures & Algorithms/special-array-i/submission-0.py
class Solution:
    def isArraySpecial(self, nums: List[int]) -> bool:
        i=0
        if len(nums)==1:
            return True
        while i<=len(nums)-2:
            if(abs(nums[i]-(nums[i+1]))%2==0):
                return False
            i+=1
        return True

        